tests/unit/plugins/module_utils/test_terraform_commands.py

```diff
@@ -0,0 +1,150 @@
+            "-target",
+            ".",
+            ".",
+        ]
+
+        self.mock.assert_called_with(*expected_cmd, check_rc=False)
+
+    # Test init method; NOT __init__
+    def test_init(self):
+        self.tf._run = self.mock
+        self.tf.init(
+            backend_config={"test_val": "test"},
+            backend_config_files=["."],
+            reconfigure=True,
+            upgrade=True,
+            plugin_paths=["."],
+        )
+
+        expected_cmd = [
+            "init",
+            "-input=false",
+            "-no-color",
+            "-backend-config",
+            "test_val=test",
+            "-backend-config",
+            ".",
+            "-reconfigure",
+            "-upgrade",
+            "-plugin-dir",
+            ".",
+        ]
+
+        self.mock.assert_called_with(*expected_cmd, check_rc=True)
+
+    def test_plan(self):
+        self.mock.return_value = (0, "", "")
+        self.tf._run = self.mock
+        self.tf.plan(target_plan_file_path=".", targets=["."], destroy=True, state_args=[""], variables_args=[""])
+
+        expected_cmd = [
+            "plan",
+            "-lock=true",
+            "-input=false",
+            "-no-color",
+            "-detailed-exitcode",
+            "-out",
+            ".",
+            "-target",
+            ".",
+            "-destroy",
+            "",
+            "",
+        ]
+
+        self.mock.assert_called_with(*expected_cmd, check_rc=False)
+
+    def test_providers_schema(self):
+        self.mock.return_value = (0, '{"format_version":"1.0"}', "")
+        self.tf._run = self.mock
+        self.tf.providers_schema()
+
+        expected_cmd = ["providers", "schema", "-json"]
+
+        self.mock.assert_called_with(*expected_cmd, check_rc=False)
+
+    def test_show(self):
+        self.mock.return_value = (0, '{"format_version":"1.0"}', "")
+        self.tf._run = self.mock
+        self.tf.show(".")
+
+        expected_cmd = ["show", "-json", "."]
+
+        self.mock.assert_called_with(*expected_cmd, check_rc=False)
+
+    def test_validate(self):
+        self.tf._run = self.mock
+        self.tf.validate(LooseVersion("0.15.0"), ["."])
+
+        expected_cmd = ["validate"]
+
+        self.mock.assert_called_with(*expected_cmd, check_rc=True)
+
+    def test_version(self):
+        extract_version = '{ \
+              "terraform_version": "1.3.6", \
+              "platform": "linux_amd64", \
+              "provider_selections": {}, \
+              "terraform_outdated": true \
+            }'
+        self.mock.return_value = (0, extract_version, "")
+        self.tf._run = self.mock
+
+        self.tf.version()
+
+        expected_cmd = ["version", "-json"]
+        self.mock.assert_called_with(*expected_cmd, check_rc=True)
+
+    def test_workspace(self):
+        self.tf._run = self.mock
+        self.tf.workspace(self.mock.WorkspaceCommand.NEW, ".")
+        expected_cmd = ["workspace", self.mock.WorkspaceCommand.NEW.value, "."]
```

Comment:
Use `WorkspaceCommand` members directly and compare with a hardcoded string, not the enum member's value.

I'd parametrize this test with [pytest.mark.parametrize](https://docs.pytest.org/en/6.2.x/parametrize.html#parametrizemark) for all three existing `WorkspaceCommand`s so we can test whether each command results in the correct `terraform` call.